### PR TITLE
Deduplicating crate dependencies (part 3 of n)

### PR DESCRIPTION
**Pull request by shergin**
The change includes:
```
tiny-keccak
tokio-timer
textwrap
winapi
rand
rand_xorshift
scopeguard
url
```
#11468

### CODE REVIEW

File: util/keccak-hasher/src/lib.rs [hunk 1]
```diff
@@ -31,8 +31,11 @@ impl Hasher for KeccakHasher {
 	type StdHasher = PlainHasher;
 	const LENGTH: usize = 32;
 	fn hash(x: &[u8]) -> Self::Out {
+		use tiny_keccak::Hasher;
```
Comment (dvdplm) [inline util/keccak-hasher/src/lib.rs:34]: Why not `use tiny_keccak::{Keccak, Hasher};` on line 23?

**Comment by ordian** [comment]
> LGTM. I'm surprised it was so easy to upgrade `tiny_keccak`; for some reason I thought it would have been much more busy work. I guess most uses are in the `parity-common` crates.

tiny-keccak 1.5.0 is still there (used in ethbloom, keccak-hash, keccak-hasher, parity-secretstore)

File: rpc/src/v1/helpers/secretstore.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ use jsonrpc_core::Error;
 use v1::helpers::errors;
 use v1::types::EncryptedDocumentKey;
 use tiny_keccak::Keccak;
+use tiny_keccak::Hasher;
```
Comment (ordian) [inline rpc/src/v1/helpers/secretstore.rs:27]: would be nice to group the imports
```suggestion
use tiny_keccak::{Keccak, Hasher};
```

File: util/network-devp2p/src/connection.rs [hunk 3]
```diff
@@ -32,6 +32,7 @@ use mio::tcp::TcpStream;
 use parity_bytes::Bytes;
 use rlp::{Rlp, RlpStream};
 use tiny_keccak::Keccak;
+use tiny_keccak::Hasher;
```
Comment (ordian) [inline util/network-devp2p/src/connection.rs:35]:
```suggestion
use tiny_keccak::{Keccak, Hasher};
```

File: util/keccak-hasher/src/lib.rs (hunk 1 above)
Comment (shergin) [reply util/keccak-hasher/src/lib.rs:34]: I am n00b in Rust. I tried that originally and got compiler error saying something like "using trait implementations ... blah blah... must be in the scope", so I moved that to the scope and the error is gone. I will investigate why exactly (but it seems correct, `¯\_(ツ)_/¯`).

**Comment by shergin** [comment]
From my understanding, most of remaining duplicating dependencies are in 3rd-party deps (or maybe I am missing something), not sure what we should/can do with that.

**Comment by ordian** [comment]
> From my understanding, most of remaining duplicating dependencies are in 3rd-party deps (or maybe I am missing something), not sure what we should/can do with that.

We still can deduplicate crates like syn and proc-macro2 by upgrading dependencies (ethabi, logos and validator -derive) and tiny-keccak (see the list above).

**Comment by shergin** [comment]
Thank you folks!
I will try to upgrade deps in a separate PR.

**Pull request merged**